### ISSUE TITLE
Argument groups not work as expected

### ISSUE THREAD

**Issue by moscartong** [joebain/args.js#18]
Hi. I have the following function:
```JavaScript
function show() {
    var args = Args([
        [
            {element:Args.OBJECT, _type:Element},
            {message:Args.STRING}
        ],
        {title:Args.STRING|Args.Optional},
        {fixed:Args.BOOL|Args.Optional, _default:false}
    ], arguments);

    //show something ...
}
```

And I expect all these should work:
```JavaScript
show('message', 'title', true); //works
show('message', 'title'); //works
show('message'); //works too :D

var el = document.createElement('div');
show(el); //not working :(
```

The last line of code throw this error:
```
Uncaught Error: Argument 0 is null or undefined but it must be not null.
```

I am using the latest version of args-js (0.10.7).

**Comment by joebain** [comment]
Hi moscartong, thanks for the report. I began looking at it and I've managed to boil it down to the following test:

```javascript
it("should pick up an Element and ignore an optional arg named title", function() {
    var el = document.createElement('div');
    var testArgs = [el];

    var args = Args([
        {element: Args.OBJECT, _type:Element},
        {title: Args.STRING | Args.Optional}
    ], testArgs);

    assert.equal(args.element, el);
    assert.equal(args.title, undefined);
});
```

Interestingly the test passes with any of the following changes:
 * The type is not `Element` but something else
 * The second argument is not present
 * The second argument is not called "title".

I suspect that it's something to do with Element types having a `title` property, I need to look more into this to discover the solution. However in the short term you should be able to solve your problem by using a different argument name, something other than title.

**Comment by moscartong** [comment]
@joebain Thanks for the quick help. Indeed it is because of the argument name "title". I change it to something else and it works fine now :+1:

**Comment by joebain** [comment]
This should now be fixed properly in 0.10.9.
